src/kernels/UniversalTranspose.cl

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2021 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "float_types.h"
+
+__kernel void UniversalTranspose(const __global _FLOAT* in,
```

Comment:
Could ```_FLOAT ``` be either ```float``` or ```half```? I think this kernel might also be applicable for ```double``` and ```int/short/char/```. Could it be suitable for all datatype?

Reply:
I guess so. I have just used a common header: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/src/kernels/float_types.h. I am not a pro in kernel development, but I could probably do that. Also, this kernel is kind of a stub. It is not performant, it just allows us to use the API for previously unsupported configurations and to implement a performant specific solution by just making a tiny, specialized transpose pseudo-solver, e.g. NCHW to NHWC.
It would probably make more sense for someone more experienced to look at the kernel and update it if needed. Maybe in a separate PR as it is not mandatory for other work to do with this code.

Reply:
Agreed, maybe in a separate PR.